pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -746,11 +746,11 @@ func (f *fqdnController) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
 func (f *fqdnController) handlePacketIn(pktIn *ofctrl.PacketIn) error {
 	klog.V(4).InfoS("Received a packetIn for DNS response")
 	waitCh := make(chan error, 1)
-	handleUDPData := func(dnsPkt *protocol.UDP) {
-		dnsData := dnsPkt.Data
+	handleDNSData := func(dnsData []byte) {
 		dnsMsg := dns.Msg{}
 		if err := dnsMsg.Unpack(dnsData); err != nil {
-			waitCh <- err
+			// A non-DNS response packet is received. Forward it to the Pod.
```

Comment:
Yes, before I made this change because I want to avoid the unexpected drop if the packet is not a DNS response. And I use whether the packet could be unpacked to determine if it's a DNS response. 
But your concern makes sense. I'm wondering if it is divided into multiple packets, will those packets be reassembled before entering the OVS?